com.unity.sg2/Editor/GraphDeltaRegistry/PreviewManager/PreviewService.cs

```diff
@@ -716,31 +777,24 @@ void AddMainPreviewData(string contextNodeName, Vector2 mainPreviewSize)
             m_CachedPreviewData.Add(m_OutputContextNodeName, m_MainPreviewData);
         }
 
-        static Shader MakeShader(string input)
-        {
-            bool tmp = ShaderUtil.allowAsyncCompilation;
-            ShaderUtil.allowAsyncCompilation = false;
-            Shader output = ShaderUtil.CreateShaderAsset(input, true);
-            ShaderUtil.allowAsyncCompilation = tmp;
-            return output;
-        }
-
         Shader GetNodeShaderObject(NodeHandler nodeReader)
         {
             string shaderOutput = Interpreter.GetShaderForNode(nodeReader, m_GraphHandle, m_RegistryInstance, out m_CachedPreviewData[nodeReader.ID.LocalPath].defaultTextures, m_Target);
             var throwAway = new List<(string, Texture)>(); // gross.
             m_CachedPreviewData[nodeReader.ID.LocalPath].shaderString = shaderOutput;
             m_CachedPreviewData[nodeReader.ID.LocalPath].blockString = Interpreter.GetBlockCode(nodeReader, m_GraphHandle, m_RegistryInstance, ref throwAway);
             m_CachedPreviewData[nodeReader.ID.LocalPath].functionString = Interpreter.GetFunctionCode(nodeReader, m_RegistryInstance);
-            return MakeShader(shaderOutput);
+            // create the shader asset without compiling the shader
+            return ShaderUtil.CreateShaderAsset(shaderOutput, false);
```

Comment:
Do tests that rely on previews still pass? For supporting tests and other inline usage, having a non-async mode or flag might be needed.

Reply:
That is true! I don't know what facilities are provided by the headless runner. The `EditorApplication` is available and accepts registration for `update` but it seems that the calls don't happen. 

My bad assumption here. Good catch. Thank you!

Reply:
From @Nightmask3 , tests are not working right now and we need to fix asap.

Reply:
Sorry for overlooking this--

It turns out, the request for a new preview material is not asynchronous, so the graphics system will block until the shader is compiled before creating the render. This is why the tests are still passing (because they ask for it immediately), but it also means you won't see a speed-up in the editor because it's probably still blocking on shader compilation (because the tool is getting the textures back immediately).

I believe this is because the `PreviewUpdateDispatcher` is requesting the textures via `PreviewService.RequestNodePreviewUpdate`, which calls `PreviewService.UpdateRenderData` to create the texture that's later pushed into the `PreviewUpdateReceiver.UpdatePreviewData`. The `UpdateRenderData` call inside `RequestNodePreviewUpdate` needs to wait until compilation is finished before being called-- so it needs to be processed as a part of the `UpdateHandler`. As it is now; it will try and create the texture immediately, which requires that the Graphics system block until the shader is compiled.